core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/verticalscale/CoreVerticalScaleActions.java

```diff
@@ -36,18 +36,28 @@
 import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
 import com.sequenceiq.cloudbreak.core.flow2.stack.StackFailureContext;
 import com.sequenceiq.cloudbreak.domain.Resource;
+import com.sequenceiq.cloudbreak.dto.InstanceGroupDto;
 import com.sequenceiq.cloudbreak.dto.StackDto;
 import com.sequenceiq.cloudbreak.reactor.api.event.StackEvent;
 import com.sequenceiq.cloudbreak.reactor.api.event.StackFailureEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.resource.CoreVerticalScalePreparationRequest;
+import com.sequenceiq.cloudbreak.reactor.api.event.resource.CoreVerticalScalePreparationResult;
 import com.sequenceiq.cloudbreak.reactor.api.event.resource.CoreVerticalScaleRequest;
 import com.sequenceiq.cloudbreak.reactor.api.event.resource.CoreVerticalScaleResult;
 import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
 import com.sequenceiq.cloudbreak.util.StackUtil;
+import com.sequenceiq.common.api.type.InstanceGroupType;
 
 @Configuration
 public class CoreVerticalScaleActions {
     private static final Logger LOGGER = LoggerFactory.getLogger(CoreVerticalScaleActions.class);
 
+    private static final String STACK_VERTICAL_SCALE_REQUEST = "StackVerticalScaleRequest";
+
+    private static final String STACK_DTO = "StackDto";
+
+    private static final String INSTANCE_GROUP = "InstanceGroup";
```

Comment:
No, what I rather meant is they were not needed if the `StackDto` (and indirectly `InstanceGroupDto`) was looked up in every flow step instead of passing it around as a flow variable. Leaving this open until an ultimate decision is made for those 2 comments above. :smile:

Reply:
@doktoric please comment here as well

Reply:
lets follow @lajosrodek suggestion.